src/server.c

```diff
@@ -5885,23 +5942,16 @@ sds genRedisInfoString(dict *section_dict, int all_sections, int everything) {
         "# Cluster\r\n"
         "cluster_enabled:%d\r\n",
         server.cluster_enabled);
+        if(server.cluster_enabled) {
+            info = genRedisInfoStringCluster(info);
+        }
```

Comment:
How fast is this command? It does an iteration of at least all slots, so it might be kind of slow? I don't know how it compares to the redis of the server though.

Reply:
the normal INFO command only has a loop on all replicas (IIRC that's the only loop).
well, that and the loop on the commands (only ones that were actually used).

so this would certainly be bad to introduce there.
maybe it could be added as a non-all section, i.e. only present if explicitly asked for.

the question is how users use this info.
i.e. the advantage of adding it to INFO only when explicitly asked for, vs having a dedicated command, is that maybe some standard tools that extract metric from INFO would be able to use it, but it is unlikely that these tools would like any per-slot information, so i'm guessing there's no reason to bring that data into the INFO command.

Reply:
ohh, sorry for being sloppy and responding without reading the code.
there's no per-slot info, just some counters.
in that case maybe they should be computed periodically / incrementally, like we do for client memory usage?

Reply:
Agree. I like the direction, I just wanted to improve the speed if it was slow for some reason.

Reply:
Hi Oran and Madolson, I would do a performance test, and check if when the cluster has a lot of nodes, how this info command speed and let you know the result. Thanks

Reply:
@hwware Any update here?

Reply:
> @hwware Any update here?

Hi Madolson, so far, I have no time working this performance test. I will let you know when I have the result,  sorry for this.